CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -396,16 +410,53 @@ public static function postProcess(&$form) {
       $html[] = $tokenHtml;
     }
 
+    $saveToFile = count($activityIds) === 1 && Civi::settings()->get('recordGeneratedLetters') === 'combined-attached';
```

Comment:
Thanks for commenting on that. I'm gonna change it because these two conditionals are logically different:

 * One interprets a *policy* (e.g. do our settings apply).
 * Another verifies a *pre-condition* assumed by the policy. However, if the pre-condition isn't met, we should probably throw an error.